includes/html/api_functions.inc.php

```diff
@@ -1905,6 +1905,12 @@ function update_device(Illuminate\Http\Request $request)
     $hostname = $request->route('hostname');
     // use hostname as device_id if it's all digits
     $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+
+    // Check that the device exists before updating fields
+    if (Device::where('device_id', $device_id)->doesntExist()) {
```

Comment:
Use DeviceCache, because the device should already be loaded by the above code.

```suggestion
    if (! DeviceCache::get($device_id)->exists) {
```

Reply:
Thanks for the suggestion. Just tested this quickly and it seems to be working.